neutron/agent/linux/iptables_manager.py

```diff
@@ -199,6 +231,35 @@ def remove_chain(self, name, wrap=True):
         self.rules = [r for r in self.rules
                       if jump_snippet not in r.rule]
 
+    def remove_chain_by_tag(self, name, tag, wrap=True):
```

Comment:
Outside of the unit tests it is called twice in IptablesMeteringDriver _process_disassociate_metering_label(). I should have looked before this. I don't know if those two uses could be converted to the new method but I'd expect so.

If I remove one of the remove chain methods then add_rule() should be changed to force the use of a tag. This would help insure users remain aware that they must maintain the tag. There is a case of iptables_manger owning a chain and iptables_firewall adding/removing a rule from that chain. Thus iptables_firewall adding a tag to that rule doesn't really make sense. The chain impacted by this situation is never deleted.

What do you suggest I do?